test/library/standard/Set/setOperatorOrderTest.chpl

```diff
@@ -0,0 +1,43 @@
+use Set;
+
+record R {
+  var a: string;
+  
+  operator ==(a:R, b:R) {
+    return true;
+  }
+
+  // We need to ensure that objects of this type are
+  // not only `==` equivalent, but also hash to the
+  // same value, based on the internal hashtable
+  // implementation.
+  proc hash(): uint {
```

Comment:
I'd add a comment either here or near the "LHS other" and "RHS other" elements explicitly noting that while they are `==` to each other, they don't hash to the same location so are independent.